### PR TITLE
Outbox proof

### PR DESCRIPTION
**Pull request by hkalodner**
- Add support for outbox proofs
- Add integration test for L2 to L1 transactions

### CODE REVIEW

File: packages/arb-evm/evm/merkleRootResult.go [hunk 1]
```diff
@@ -136,9 +182,47 @@ func newMerkleTreeFromValue(val value.Value) (MerkleNode, error) {
...
 	}
 }
+
+type MerkleRootProof struct {
+	Nodes []common.Hash
+	Path  []bool
+	Data  []byte
+}
+
+func (m *MerkleRootResult) GenerateProof(index uint64) (*MerkleRootProof, error) {
```
Comment (PlasmaPower) [inline packages/arb-evm/evm/merkleRootResult.go:199]: We already have an existing merkle tree and proof generation system somewhere in the ethbridge MerkleTree, is that fundamentally incompatible with this in some way? I'd prefer to only have one merkle tree system in Go.

Comment (hkalodner) [reply packages/arb-evm/evm/merkleRootResult.go:199]: Agreed that it'd be nice to limit these. The difference is basically that this tree is produced by ArbOS using a different structure and unifying them would require significantly altering ArbOS

Comment (PlasmaPower) [reply packages/arb-evm/evm/merkleRootResult.go:199]: Ah, I was wondering about that. Makes sense to keep it as-is then.

Comment (hkalodner) [reply packages/arb-evm/evm/merkleRootResult.go:199]: At least we use the same prover on the solidity side which minimizes the issues there and means we stick to a unified proving strategy

File: packages/arb-avm-cpp/cmachine/nodestorage_test.go [hunk 2]
```diff
@@ -52,12 +52,11 @@ func TestMessageBatch(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	logIndex, err := nodeStore.GetMessageBatch(testBatchNumber)
-	if err != nil {
-		logger.Error().Stack().Err(err).Send()
+	logIndex := nodeStore.GetMessageBatch(testBatchNumber)
+	if logIndex == nil {
 		t.Fatal(err)
```
Comment (PlasmaPower) [inline packages/arb-avm-cpp/cmachine/nodestorage_test.go:57]: This should probably be `t.Fatal("GetMessageBatch failed")` or something

File: packages/arb-util/arbos/arbos.go [hunk 3]
```diff
@@ -16,6 +16,8 @@ var ARB_BLS_ADDRESS = ethcommon.HexToAddress("0x00000000000000000000000000000000
 var ARB_FUNCTION_TABLE_ADDRESS = ethcommon.HexToAddress("0x0000000000000000000000000000000000000068")
 var ARB_OWNER_ADDRESS = ethcommon.HexToAddress("0x000000000000000000000000000000000000006B")
 
+var ARB_NODE_INTERFACE_ADDRESS = ethcommon.HexToAddress("0x000000000000000000000000000000000000006E")
```
Comment (PlasmaPower) [inline packages/arb-util/arbos/arbos.go:19]: I'd put this at a much higher address to differentiate it and not conflict with future arbos "contracts"

**Review by PlasmaPower — CHANGES_REQUESTED** [review]

File: packages/arb-util/arbos/arbos.go (hunk 3 above)
Comment (hkalodner) [reply packages/arb-util/arbos/arbos.go:19]: Agree with this, but we'll need to charge ArbOS. I'll flag @edfelten on this question

Comment (hkalodner) [reply packages/arb-util/arbos/arbos.go]: This was changed to 200 to keep it near the arbos address block, but leave plenty of room for future expansion

**Review by PlasmaPower — APPROVED** [review]
LGTM

**Pull request merged**